assets/src/edit-story/app/media/utils/getResourceFromMedia3p.js

```diff
@@ -245,8 +301,14 @@ function getGifResourceFromMedia3p(m) {
     alt: null,
     local: false,
     sizes: imageUrls,
+    output: {
+      mimeType: videoUrls.mp4.full.mime_type,
+      sizes: videoUrls,
+      src: videoUrls.mp4.full.source_url,
+    },
```

Comment:
This doesn't match `StoryPropTypes.gifResource`?

Reply:
for the `output` property? I have this:
```
  output: PropTypes.shape({
    mimeType: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired,
  }),
```

It is missing the `sizes` property, but so were `imageResource` and `videoResource`. I can add `sizes` to `types` for clarity (or remove it since it's not being used). Or are you referring to something else? I didn't see a prop-type error during testing but I might have missed something. :)

Also, how do you feel about this data structure? I'm having trouble coming up with ways to improve it. For example, I don't like that it's not very flat, and I'm not sure of the best way to deliver the "video" resources from Tenor for the GifOutput without adding a new optional property to `element.resource`.

Reply:
Yeah passing props that aren't declared in proptypes won't cause warnings. But if `sizes` is used in these places, it's better to define proptypes for it.

> Also, how do you feel about this data structure? I'm having trouble coming up with ways to improve it.

Same here. It seems fine. We can always change the data structure if needed and add a migration for it. For example should we ever want to use both webm and mp4 for output, this would need to become an array of output formats